### ISSUE TITLE
Question about running with custom dataset

### ISSUE THREAD

**Issue by chqzeng** [vLAR-group/GrowSP#10]
Hi this is a unique project as it is an unsupervised approach,  the result looks promising, great work! I have some questions and really appreciate if you can help to provide some suggestions:

- I have some point cloud in (x,y,z, intensity) format, Is it possible I can run my custom data using this project? if yes, which script should I start with?  should I prepare some thing and run scripts under `data_prepare` as well?

- I assume this unsupervised model, essentially cluster approach, but then what is the meaning of `training` in this project ?   Can I directly run my dataset without training?  what is the extra value of do the training here?

- I successfully run : ``CUDA_VISIBLE_DEVICES=0, python train_SemanticKITTI.py`  , how to interpret the output?  where the result files stored ?

Thank you, 
Chui,

**Comment by zhang-zihui** [comment]
Hi, @chqzeng 
Sounds like your data are similar to SemanticKITTI. We don't provide a specific script for custom data, but you can refer to our codes used in SemanticKITTI. You also need to construct superpoints similar to the codes under "data_prepare."

Our unsupervised method relies on deep networks, so it requires training. We get the pseudo labels with the help of K-Means, then train the deep networks with their supervision to enhance the discrimination of the network's features. Consequently, the features will be increasingly discriminative and clearly better than a random initialized network. So, running without training doesn't work. And I recommend reading our paper at first, as it will facilitate a better understanding of the methodology.

The output should be a folder containing the checkpoints as well as a log file. This log file records training loss, the k-means error, the accuracy of superpoints and primitives, and segmentation results on the validation set. You can find them in our paper.
